osu.Game/Skinning/LegacyKeyCounterDisplay.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Screens.Play.HUD;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osuTK;
+
+namespace osu.Game.Skinning
+{
+    public partial class LegacyKeyCounterDisplay : KeyCounterDisplay
+    {
+        private static readonly Colour4 active_colour_top = Colour4.FromHex(@"#ffde00");
+        private static readonly Colour4 active_colour_bottom = Colour4.FromHex(@"#f8009e");
+
+        protected override FillFlowContainer<KeyCounter> KeyFlow { get; }
+
+        private readonly Sprite backgroundSprite;
+
+        public LegacyKeyCounterDisplay()
+        {
+            AutoSizeAxes = Axes.Both;
+
+            AddRangeInternal(new Drawable[]
+            {
+                backgroundSprite = new Sprite
```

Comment:
Because the background isn't contained within `KeyFlow`, the background will show regardless of the "always show key overlay" setting:

![osu! 2024-08-06 at 05 42 11](https://github.com/user-attachments/assets/1fe0ecdd-ba9b-42a9-9e24-8f44342f4b5f)

This is due to 

https://github.com/ppy/osu/blob/798e677c092a4ad02e0fe0a81163c14b57ff90d4/osu.Game/Screens/Play/HUD/KeyCounterDisplay.cs#L37-L47

Should be simple enough to fix, I may look at it subsequently

Reply:
Fixed in https://github.com/peppy/osu/commit/8619bbb9435c82018ffc50ab15a61430ae77146a.